### PR TITLE
Record deleted(by mod) status to prevent re-appear

### PR DESCRIPTION
**Pull request by tribela**
Related: #10731 

Like Tombstone, deleted status by moderation should not be re-appeared on timeline.

### CODE REVIEW

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
`db/schema.rb` is missing the new table, and `app/models/redacted_status.rb` is missing the annotations.

Also, I wonder if it wouldn't be better to add a flag in the `tombstones` table instead, to perform only one lookup instead of two.

**Comment by Kjwon15** [comment]
Thanks for comment. I agree that using tombstone instead of making another table is performance better.
But that status is not actually deleted. So I am curious it is fine to add to Tombstone.

**Comment by ThibG** [comment]
Yes, I think I'd add a field to `Tombstone` to mark that it was manually discarded by an admin rather than actually deleted.

**Comment by Kjwon15** [comment]
Oh that's seems good.
Anyway, I am not rails expert. Can you tell me how to add annotation comments?

**Comment by ThibG** [comment]
They are added when running the migration in the development environment. Usually I add them manually 🤷

**Comment by Kjwon15** [comment]
Thanks!
looks fine now.

File: db/schema.rb [hunk 1]
```diff
@@ -591,8 +591,8 @@
   create_table "status_pins", force: :cascade do |t|
     t.bigint "account_id", null: false
     t.bigint "status_id", null: false
-    t.datetime "created_at", default: -> { "now()" }, null: false
-    t.datetime "updated_at", default: -> { "now()" }, null: false
+    t.datetime "created_at", default: -> { "CURRENT_TIMESTAMP" }, null: false
+    t.datetime "updated_at", default: -> { "CURRENT_TIMESTAMP" }, null: false
     t.index ["account_id", "status_id"], name: "index_status_pins_on_account_id_and_status_id", unique: true
```
Comment (ThibG) [inline db/schema.rb]: I am not sure why these changes were made?

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Also, it looks like the migration script is missing, now.

File: db/schema.rb (hunk 1 above)
Comment (Kjwon15) [reply db/schema.rb]: IDK too.. But db:migrate makes that.

**Review by ClearlyClaire — APPROVED** [review]
